### PR TITLE
mcap: add version 0.1.0

### PR DESCRIPTION
**Pull request by jtbandes**
Hello 👋

I'm part of a team developing a file format, [MCAP](https://github.com/foxglove/mcap), and we would like the C++ reader/writer libraries to be available on ConanCenter. This is the first published version of the package, which is a header-only library, with dependencies on `fmt`, `lz4`, and `zstd`.

Closes https://github.com/foxglove/mcap/issues/201

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/mcap/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    description = "A C++ implementation of the MCAP file format"
+    license = "Apache-2.0"
+    topics = ("mcap", "serialization", "deserialization", "recording")
+
+    settings = ("os", "compiler", "build_type", "arch")
+    requires = ("fmt/8.1.1", "lz4/1.9.3", "zstd/1.5.2")
+    generators = ("cmake", "cmake_find_package")
+
+    _source_root = "source_root"
+    _source_package_path = os.path.join(_source_root, "cpp", "mcap")
```
Comment (uilianries) [inline recipes/mcap/all/conanfile.py:19]:
```suggestion
    @property
    def _source_subfolder(self):
        return "source_subfolder"
        
    @property
    def _source_package_path(self):
        return os.path.join(self._source_subfolder, "cpp", "mcap")
```

Try to use common patterns from Conan Center index. It makes easy to review and compare recipes.

File: recipes/mcap/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_root)
```
Comment (uilianries) [inline recipes/mcap/all/conanfile.py:22]:
```suggestion
        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Comment by jtbandes** [comment]
Hi, is there any update on this, or anything else I can do to help shepherd it along?

**Comment by uilianries** [comment]
@danimtb could you please review this PR, so we can merge it.

**Review by prince-chrismc — APPROVED** [review]

File: recipes/mcap/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "17")
+        if self.settings.compiler == "apple-clang" and tools.Version(self.settings.compiler.version) <= 11:
+            raise ConanInvalidConfiguration("Compiler version is not supported, c++17 support is required")
+        if (self.settings.compiler == "gcc" or self.settings.compiler == "clang") and tools.Version(self.settings.compiler.version) <= 8:
+            raise ConanInvalidConfiguration("Compiler version is not supported, c++17 support is required")
+        if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) <= "16.8":
```
Comment (SpaceIm) [inline recipes/mcap/all/conanfile.py:36]: don't Visual Studio version like that, it raises for Visual Studio 2019.

Comment (jtbandes) [reply recipes/mcap/all/conanfile.py:36]: I got this from https://github.com/conan-io/conan/issues/8002, which is linked from the readme: https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#why-is-a-toolscheck_min_cppstd-call-not-enough, via this conversation on another package I worked on: https://github.com/conan-io/conan-center-index/pull/8662#discussion_r782533406  cc @prince-chrismc

**Pull request merged**